medusa/indexers/tvdbv2/tvdbv2_api.py

```diff
@@ -157,12 +157,28 @@ def search(self, series):
         """
         log.debug('Searching for show: {0}', series)
 
+        results = None
+        # If search term is digit's only, store it and attempt to search by id.
+        show_by_id = None
+
+        if series.isdigit():
+            try:
+                show_by_id = self._get_show_by_id(series, request_language=self.config['language'])
+            except (IndexerShowNotFound, IndexerShowNotFoundInLanguage):
+                pass
+
         results = self._show_search(series, request_language=self.config['language'])
-        if not results:
+        if not results and not show_by_id:
             return
 
-        mapped_results = self._object_to_dict(results, self.series_map, '|')
-        mapped_results = [mapped_results] if not isinstance(mapped_results, list) else mapped_results
+        mapped_results = []
+        if results:
+            mapped_results = self._object_to_dict(results, self.series_map, '|')
+            mapped_results = [mapped_results] if not isinstance(mapped_results, list) else mapped_results
```

Comment:
Can we standardize this to use `_map_results` like the other indexers? Possibly in `_show_search` (aka `_get_show_by_name`)

Reply:
I'm not sure. Ill check out the effort